src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let subSeparator = separator;
+
+    if (separator === null) {
+      subSeparator = 'null';
+    }
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += subSeparator;
+      } else {
+        result += this[i] + subSeparator;
```

Comment:
```suggestion
        result += this[i] + '' + subSeparator;
```
Якщо записати так, то можна позбавитись від

```
if (separator === null) {
      subSeparator = 'null';
    }
```